src/log_unstable.rs

```diff
@@ -64,16 +64,13 @@ impl Unstable {
     pub fn maybe_first_index(&self) -> Option<u64> {
         self.snapshot
             .as_ref()
-            .map(|snap| snap.get_metadata().get_index() + 1)
+            .map(|snap| snap.get_metadata().index + 1)
     }
 
     /// Returns the last index if it has at least one unstable entry or snapshot.
     pub fn maybe_last_index(&self) -> Option<u64> {
         match self.entries.len() {
-            0 => self
-                .snapshot
-                .as_ref()
-                .map(|snap| snap.get_metadata().get_index()),
+            0 => self.snapshot.as_ref().map(|snap| snap.get_metadata().index),
```

Comment:
do we still need to call `get_metadata()`?

Reply:
`.metadata` returns `Option`, `.get_metadata` will return default value if `None`

Reply:
However, seems like there's a `metadata()` function.

Reply:
No, there isn't.